src/combinatoric/combinatoric.ts

```diff
@@ -78,4 +81,9 @@ export const structureHelpersBy = <TStructure extends CombinatoricStructureType>
 	return { make: make as never as Make<TStructure, 0>, strict: make, isSpecimen, rangeWith, iterateBy };
 };
 
-export const extractValues = <T>(collection: Collection<T>) => (Array.isArray(collection) ? collection : Object.values(collection));
+export const extractValues = <T>(collection: Collection<T>): Iterable<T> => {
+	if (Array.isArray(collection)) return collection;
+	if (Lazy.isArray(collection)) return collection;
```

Comment:
would it be bad if Lazy.isArray would wrap around the case where `Array.isArray` is true?

Reply:
Yeah, because it has a return type of `maybe is ILazyArray<T>`, instead of just `boolean`. So it wouldn't be true if it were inclusive of `Array.isArray`.